### PR TITLE
Add hacky via support for RGB Matrix

### PR DESCRIPTION
**Pull request by drashna**
## Description

Adds rgb matrix functions for via's lighting system, rather than the hacky rgblight stuff. 

This only adds support on qmk side, as via is developed independently of qmk. 

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

File: quantum/via.c [hunk 1]
```diff
@@ -495,3 +514,86 @@ void via_qmk_rgblight_set_value(uint8_t *data) {
...
+        case 2:
+        default:
+            return RGBLIGHT_SPEED_2;
+        case 3:
+            return RGBLIGHT_SPEED_3;
+    }
+}
+
+static uint8_t speed_to_rgblight(uint8_t rgb_matrix_speed) {
+    if (rgb_matrix_speed < ((RGBLIGHT_SPEED_0 + RGBLIGHT_SPEED_1) / 2)) {
```
Comment (yanfali) [inline quantum/via.c:544]: this looks like it could be converted into a lookup table

File: quantum/via.c [hunk 2]
```diff
@@ -495,3 +514,86 @@ void via_qmk_rgblight_set_value(uint8_t *data) {
...
+            value_data[0] = rgb_matrix_get_mode();
+            break;
+        case id_qmk_rgblight_effect_speed:
+            value_data[0] = speed_to_rgblight(rgb_matrix_get_speed());
+            break;
+        case id_qmk_rgblight_color:
+            value_data[0] = rgb_matrix_get_hue();
+            value_data[1] = rgb_matrix_get_sat();
+            break;
+    }
```
Comment (yanfali) [inline quantum/via.c:572]: no default?

File: quantum/via.c [hunk 3]
```diff
@@ -495,3 +514,86 @@ void via_qmk_rgblight_set_value(uint8_t *data) {
...
+                rgb_matrix_enable_noeeprom();
+            }
+            break;
+        case id_qmk_rgblight_effect_speed:
+            rgb_matrix_set_speed_noeeprom(speed_from_rgblight(value_data[0]));
+            break;
+        case id_qmk_rgblight_color:
+            rgb_matrix_sethsv_noeeprom(value_data[0], value_data[1], rgb_matrix_get_val());
+            break;
+    }
```
Comment (yanfali) [inline quantum/via.c:596]: same

File: quantum/via.c (hunk 2 above)
Comment (drashna) [reply quantum/via.c:572]: It's mostly copy-pasta, and those didn't include it.  Any other values aren't technically supported, either.

But I can add a default, if need be.

File: quantum/via.c (hunk 1 above)
Comment (drashna) [reply quantum/via.c:544]: probably.  I'll just need to figure out how to set that up

File: quantum/via.c (hunk 2 above)
Comment (tzarc) [reply quantum/via.c:572]: Shouldn't matter -- the buffer is preallocated and shouldn't be messed with if unused.
All we'd end up with is `default: break;`.

File: quantum/via.c (hunk 3 above)
Comment (tzarc) [reply quantum/via.c:596]: same

File: quantum/via.c (hunk 1 above)
Comment (tzarc) [reply quantum/via.c:544]: I wouldn't be too worried, this code should be a relative throwaway once XAP lands anyway.

**Pull request merged**